src/components/OrderConfirmation/OrderConfirmation.js

```diff
@@ -0,0 +1,157 @@
+import * as React from "react";
+import dayjs from "dayjs";
+import { Link } from "gatsby";
+import PropTypes from "prop-types";
+
+import { Metadata } from "../Metadata";
+import { ActionStateIcon, ArrowRightIcon } from "../Icons";
+import { Panel } from "../Panel";
+import HighlightedLink from "../HighlightedLink";
+
+import { usePlans } from "../../contexts/plans";
+import { usePortalSettings } from "../../contexts/portal-settings";
+import accountsService from "../../services/accountsService";
+import { DATE_FORMAT } from "../../lib/config";
+
+import { PriceInfo } from "./PriceInfo";
+
+const State = {
+  Waiting: "Waiting",
+  Success: "Success",
+  Failure: "Failure",
+};
+
+const TITLE_BY_STATE = {
+  Waiting: "Processing payment",
+  Success: "Success!",
+  Failure: "Error occurred",
+};
+
+export const OrderConfirmation = ({ sessionId }) => {
+  const { settings } = usePortalSettings();
+  const { getPlanById } = usePlans();
+  const [state, setState] = React.useState(State.Waiting);
+  const [subscription, setSubscription] = React.useState(null);
+  const [isTakingTooLong, setIsTakingTooLong] = React.useState(false);
+
+  React.useEffect(() => {
+    const fetchSession = async () => {
+      try {
+        const sub = await accountsService
+          .get(`stripe/checkout/${sessionId}`, {
+            timeout: 30_000,
+            retry: 5,
+          })
+          .json();
+        setSubscription(sub);
+        setState(sub.status === "active" ? State.Success : State.Failure);
+      } catch (err) {
+        setState(State.Failure);
+        setSubscription(null);
+        console.log(err);
+      }
+    };
+
+    if (sessionId) {
+      fetchSession();
```

Comment:
Shouldn't we `await` here?

Reply:
We don't need to (`fetchSession` will update the state as it goes).

But your comment did cause me to re-think this a little bit and I decided to make all of this request-related state a custom hook, see 13bc10d